src/sql/workbench/services/query/common/queryRunner.ts

```diff
@@ -243,6 +244,9 @@ export default class QueryRunner extends Disposable {
 		// this also isn't exact but its the best we can do
 		this._queryEndTime = new Date();
 
+		perf.mark(`sql/query/${this.uri}/handleQueryComplete`);
+		this.logPerfMarks();
```

Comment:
So this means we don't spit out the perf marks until the query is complete - right? One thing to consider is that if a query is "stuck" we still may want to use the perf marks to see where it's getting stuck at. But this is a good start and should cover most of the scenarios we care about so not an immediate concern IMO.

Reply:
Yes, they are logged when the query is complete.